gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2949,6 +2947,31 @@ func checkIntegerType(xt Type) {
 	}
 }
 
+// like checkOrConvertType() but for bool type.
+func checkOrConvertBoolType(store Store, last BlockNode, x Expr) {
+	if cx, ok := x.(*ConstExpr); ok {
+		convertConst(store, last, cx, BoolType)
+	} else if nx, ok := (x).(*NameExpr); ok {
+		xt := evalStaticTypeOf(store, last, Expr(nx))
+		if dxt, ok := xt.(*DeclaredType); ok {
+			checkType(dxt.Base, BoolType, false)
+		} else {
+			panic(fmt.Sprintf("expected declared type, but got %v", xt.Kind()))
+		}
+	} else if x != nil {
```

Comment:
Could we not just use `t := evalStaticTypeOf(x); checkType(baseOf(t), BoolType, false)` (and avoid creating `checkBoolType`?)?